Documentation/api.md

```diff
@@ -0,0 +1,112 @@
+# The dex API
+
+Dex provides a [gRPC][grpc] service for programmatic modification of dex's state. The API is intended too expose hooks for management applications and is not expected to be used by most installations.
+
+This document is an overview of how to interact with the API.
+
+## Configuration
+
+Admins that wish to expose the gRPC service must add the following entry to the dex config file. This option is off by default.
+
+```
+grpc:
+  # Cannot be the same address as an HTTP(S) service.
+  addr: 127.0.0.1:5557
+  # Server certs. If TLS credentials aren't provided dex will generate self-signed ones.
+  tlsCert: /etc/dex/grpc.crt
+  tlsKey: /etc/dex/grpc.key
+```
+
+## Generating clients
+
+gRPC is a suite of tools for generating client and server bindings from a common declarative language. The canonical schema for dex's API can be found in the source tree at `api/api.proto`. Go bindings are generated and maintained in the same directory for internal use.
+
+To generate a client for your own project install [`protoc`][protoc], install a protobuf generator for your project's language, and download the `api.proto` file. An example for a Go project:
+
+```
+# Install protoc-gen-go.
```

Comment:
If these are shell commands they should begin with `$ `. If it's a script make that clear. I like annotating it like this:

`script-name.sh`:
```
# script contents
...
```

But anything that makes it clear works for me.